.changeset/great-hotels-unite.md

```diff
@@ -0,0 +1,10 @@
+---
+"@comet/cms-api": major
+---
+
+Refactor auth-decorators
+
+-   Remove `@PublicApi()`-decorator
+-   Rename `@DisableGlobalGuard()`-decorator to `@DisableCometGuards()`
+
+When using the `@DisableCometGuards()`-decorator it's not allowed to send a `x-include-invisible-content`-header.
```

Comment:
why that? isn't that a problem when you need to make requests to an api where parts have DisableCometGuards and other not and you want to configure the gql client to always send the header (as we currently do)?

Reply:
> what exactly we want to achieve here.

Example: Let's say we have a products graphql query that is used in a site with a client side component for a product search
- for admin users: in site preview where it should also show invisible products (if the user activated the toggle)
- for any (also non authed) user: the header must not allowed

(A BFF would not have this problem)